chain/chain/src/flat_storage_resharder.rs

```diff
@@ -507,29 +582,29 @@ fn copy_kv_to_child(
         return Err(Error::ReshardingError(err_msg.to_string()));
     }
     // Add the new flat store entry.
-    store_update.set(new_shard_uid, key, Some(value));
+    store_update.set(new_shard_uid, key, value);
     Ok(())
 }
 
 /// Copies a key-value pair to both children.
 fn copy_kv_to_all_children(
     status: &SplittingParentStatus,
     key: Vec<u8>,
-    value: FlatStateValue,
+    value: Option<FlatStateValue>,
     store_update: &mut FlatStoreUpdateAdapter,
 ) {
-    store_update.set(status.left_child_shard, key.clone(), Some(value.clone()));
-    store_update.set(status.right_child_shard, key, Some(value));
+    store_update.set(status.left_child_shard, key.clone(), value.clone());
+    store_update.set(status.right_child_shard, key, value);
```

Comment:
not related to this PR
It feels like the set method should only need a reference to the key and value. In theory it should serialize those and for that ref should be sufficient. Then the clone would not be needed.